docs/ibc/1_IBC_ARCHITECTURE.md

```diff
@@ -0,0 +1,141 @@
+
+Of the underlying consensus protocols and ledgers IBC requires a set of primitive functions and properties as defined in [ICS 2](../../spec/ics-2-consensus-requirements), primarily finality, cheaply-verifiable consensus transcripts, and simple key-value store functionality. Of the network infrastructure protocol layer (and physical network layer) IBC requires only eventual data delivery — no authentication, synchrony, or ordering properties are assumed.
+
+## Operation
+
+The primary purpose of IBC is to provide reliable, authenticated, ordered communication between modules running on independent host distributed ledgers. This requires protocol logic in the following areas:
+- Data relay
+- Reliability
+- Flow control
+- Authentication
+- Connections
+- Multiplexing
+
+The following paragraphs outline the protocol logic within IBC for each area.
+
+### Data relay
+
+IBC assumes the existence of a set of relayer processes with access to an underlying network protocol stack (likely TCP/IP, UDP/IP, or QUIC/IP) and physical interconnect infrastructure. These relayer processes monitor a set of ledgers implementing the IBC protocol, continuously scanning the state of each ledger and executing transactions on another ledger when outgoing datagrams have been committed. For correct operation and progress in a connection between two ledgers, IBC requires only that at least one correct and live relayer process exists which can relay between the ledgers.
+
+### Reliability
+
+The network layer and relayer processes may behave in arbitrary ways, dropping, reordering, or duplicating packets, purposely attempting to send invalid transactions, or otherwise acting Byzantine. This must not compromise the safety or liveness of IBC. This is achieved by assigning a sequence number to each packet sent over an IBC connection, which is checked by the IBC handler on the receiving ledger, and providing a method for the sending ledger to check that the receiving ledger has in fact received and handled a packet before sending more packets or taking further action. Cryptographic commitments are used to prevent datagram forgery: the sending ledger commits to outgoing datagrams, and the receiving ledger checks these commitments, so datagrams altered in transit by a relayer will be rejected.
```

Comment:
Define safety/liveness in the context of IBC

Reply:
I'm having trouble actually defining safety in IBC independently of safety in the underlying consensus engines.

Liveness in IBC seems identical to data availability.

Maybe the only parameter of IBC that matters is data availability.

Reply:
I think there is a notion of "safety, assuming correct blockchain operation, but possible non-liveness", and a separate notion of "safety, assuming Byzantine blockchain operation" - IBC needs to provide the former and cannot provide the latter in a strong sense but can expose options to applications to mitigate / contain damage from Byzantine blockchains.

Reply:
> options to applications to mitigate / contain damage

is there a list somewhere?

Reply:
> is there a list somewhere?

No, but there should be - that's a topic for https://github.com/cosmos/ics/issues/6.